source/slang/slang.cpp

```diff
@@ -3376,6 +3481,18 @@ SLANG_NO_THROW SlangResult SLANG_MCALL ComponentType::getEntryPointCode(
     return artifact->loadBlob(ArtifactKeep::Yes, outCode);
 }
 
+SLANG_NO_THROW SlangResult SLANG_MCALL ComponentType::computeDependencyBasedHash(slang::Checksum* outHashCode)
+{
+    SLANG_UNUSED(outHashCode);
+    return SLANG_E_NOT_AVAILABLE;
+}
+
+SLANG_NO_THROW SlangResult SLANG_MCALL ComponentType::computeASTBasedHash(slang::Checksum* outHashCode)
+{
+    SLANG_UNUSED(outHashCode);
+    return SLANG_E_NOT_AVAILABLE;
+}
```

Comment:
I'm not sure why these need to be implemented here at all, as opposed to being pure-virtual methods in `ComponentType`.

I'm also not clear on why these are `COM`-ish methods (with `SLANG_MCALL`, etc.) when they aren't part of any COM interface that `ComponentType` is implementing (at least AFAICT).

Reply:
COM-ish methods because they're defined under `IComponentType`, which is necessary for `RendererBase::getEntryPointCodeFromShaderCache` to call them on `program` as it's passed in as an `IComponentType`.

Reply:
Fixed locally, will be in next push